### PR TITLE
Expand select2 exposure for external customization

### PR DESCRIPTION
**Pull request by cthrax**
I thought I'd throw this over to you if you're interested, I'm using it from my repo right now. It just makes it a little easier to get the full select2 functionality, though I've seen in the issues you're mostly moving away from select2, but it might save you some issues filed.

### CODE REVIEW

File: src/aa.select2.js [hunk 1]
```diff
@@ -87,8 +87,16 @@ angular
           //run a query to get options (search)
           if (inAjaxMode) {
             derivedOpts.query = function (query) {
-              settings.options(query.term)
-                .success(function (data) {
+              // TODO: Ought to handle the failure case, instead of spinning forever
+              var queryResult = settings.options(query.term);
+              // Assume if it's not a promise and is an array, it's data
+              if (!queryResult.then && queryResult.hasOwnProperty["length"]) {
```
Comment (johnculviner) [inline src/aa.select2.js]: hasOwnProperty("length") ?

**Comment by johnculviner** [comment]
Hey thanks for the ideas, i'll give it a merge if you fix the CI issues. Thank you!
https://travis-ci.org/AngularAgility/AngularAgility/builds/60262853

File: src/aa.select2.js (hunk 1 above)
Comment (cthrax) [inline src/aa.select2.js]: Checking whether it's an array like object. Don't care if it's actually an array, as long as behaves like one.

**Comment by cthrax** [comment]
@johnculviner I think we should be all good on Travis now.

**Comment by johnculviner** [comment]
Hey thanks for updating. Don't have the best test coverage around this thing but noticed the select2 demo page is broken by this change. If you do a 

```
grunt build
```

```
node demo/server.js
```

on your current branch you can see ajax results don't come back

**Comment by cthrax** [comment]
That was a different use than I had for the ajax. Sorry I missed it. The latest should fix that. Thanks for the pointer on running the demo too.

**Comment by johnculviner** [comment]
sweet, thanks for the changes!

**Pull request merged**